### DISCUSSION TITLE
How can I enable Treesitter textobjects swap?

### DISCUSSION THREAD

**Discussion by rpbaptist** [LazyVim/LazyVim#4903] [Q&A]
I would like to enable this feature: https://github.com/nvim-treesitter/nvim-treesitter-textobjects?tab=readme-ov-file#text-objects-swap

My current treesitter config within LazyVim looks like this:

```lua
return {
  "nvim-treesitter/nvim-treesitter",
  opts = function(_, opts)
    if type(opts.ensure_installed) == "table" then
      vim.list_extend(opts.ensure_installed, {
        "git_rebase",
        "gitattributes",
        "gitcommit",
        "gitignore",
        "sql",
        "tsx",
      })
    end
  end,
}
```

I tried adding this after the if statement closes:

```lua
    table.insert(opts.textobjects, {
      swap = {
        enable = true,
        swap_next = {
          ["<C-S-n>"] = "@parameter.inner",
        },
        swap_previous = {
          ["<C-S-o>"] = "@parameter.inner",
        },
      },
    })
```

But it's not being picked up. Any advice?

**Comment by qwertzui11** [comment]
I have been able to make it work, by adapting my dotfiles, by adding:

```lua
  {
    "nvim-treesitter/nvim-treesitter-textobjects",
    keys = {
      {
        "<leader>a",
        function()
          require("nvim-treesitter-textobjects.swap").swap_next("@parameter.inner")
        end,
        desc = "Swap parameter with next",
      },
      {
        "<leader>A",
        function()
          require("nvim-treesitter-textobjects.swap").swap_previous("@parameter.inner")
        end,
        desc = "Swap parameter with previous",
      },
    },
  },
```

I think [in the current implementation](https://github.com/LazyVim/LazyVim/blob/96f4f18d7d81c786ac0df5723bc7aca058bf2165/lua/lazyvim/plugins/treesitter.lua#L169) there is no support for the `swap` object.

Which I think is fine, due to the whole `nvim-treesitter-textobjects` options being weird in the first place :smile: . Just setting the keys is according to the official [README](https://github.com/nvim-treesitter/nvim-treesitter-textobjects?tab=readme-ov-file#text-objects-swap)

Hope this helps, Cheers
